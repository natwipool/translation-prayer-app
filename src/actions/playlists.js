import uuid from 'uuid';

// ADD_PLAYLIST
export const addPlaylist = ({ name = '', lists = [] } = {}) => ({
  type: 'ADD_PLAYLIST',
  playlist: {
    id: uuid(),
    name,
    lists
  }
});

// REMOVE_PLAYLIST
export const removePlaylist = ({ id } = {}) => ({
  type: 'REMOVE_PLAYLIST',
  id
});

// EDIT_PLAYLIST
export const editPlaylist = (id, updates) => ({
  type: 'EDIT_PLAYLIST',
  id,
  updates
});
