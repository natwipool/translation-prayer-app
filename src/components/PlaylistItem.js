import React from 'react';

const PlaylistItem = (props) => (
  <div>
    <h3>{props.transPrayerList.precept}</h3>
    <p>{props.transPrayerList.category}</p>
  </div>
);

export default PlaylistItem;