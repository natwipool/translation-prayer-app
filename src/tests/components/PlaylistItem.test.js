import React from 'react';
import { shallow } from 'enzyme';
import PlaylistItem from '../../components/PlaylistItem';
import playlists from '../fixtures/playlists';

test('should render PlaylistItem correctly', () => {
  const wrapper = shallow(<PlaylistItem transPrayerList={playlists[1]} />);
  expect(wrapper).toMatchSnapshot();
});