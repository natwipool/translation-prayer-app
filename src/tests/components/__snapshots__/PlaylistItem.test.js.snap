// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PlaylistItem correctly 1`] = `
<div>
  <h3 />
  <p />
</div>
`;
